src/hb-ot-glyf-table.hh

```diff
@@ -350,58 +339,48 @@ struct glyf
       return size;
     }
 
-    // TODO rewrite using new iterator framework if possible
-    struct Iterator
+    struct iter_t
```

Comment:
I guess need some help for porting this to dagger, started with this but didn't go well
```diff
diff --git a/src/hb-ot-glyf-table.hh b/src/hb-ot-glyf-table.hh
index a02ffaf5..bcae923a 100644
--- a/src/hb-ot-glyf-table.hh
+++ b/src/hb-ot-glyf-table.hh
@@ -339,7 +339,7 @@ struct glyf
       return size;
     }
 
-    struct iter_t
+    struct iter_t : hb_iter_t<iter_t, const CompositeGlyphHeader *>
     {
       typedef const CompositeGlyphHeader *__item_t__;
       iter_t (hb_bytes_t glyph_, __item_t__ current_) : glyph (glyph_), current (current_)
diff --git a/src/hb-subset-plan.cc b/src/hb-subset-plan.cc
index 7f601b56..3c1770bd 100644
--- a/src/hb-subset-plan.cc
+++ b/src/hb-subset-plan.cc
@@ -45,12 +45,12 @@ _add_gid_and_children (const OT::glyf::accelerator_t &glyf,
 
   hb_set_add (gids_to_retain, gid);
 
-  auto iter = glyf.get_composite_iter (gid);
-  while (iter.__more__ ())
-  {
-    _add_gid_and_children (glyf, (hb_codepoint_t) iter.__item__ ()->glyphIndex, gids_to_retain);
-    iter.__next__ ();
-  }
+  + glyf.get_composite_iter (gid)
+  | hb_map ([&] (const OT::glyf::CompositeGlyphHeader *item)
+           {
+             _add_gid_and_children (glyf, (hb_codepoint_t) item->glyphIndex, gids_to_retain);
+           })
+  ;
 }
 
 #ifndef HB_NO_SUBSET_CFF
```